### ISSUE TITLE
Replace boolean flags on `Texture::from_image` with `TextureSettings` struct

### ISSUE THREAD

**Issue by bvssvni** [PistonDevelopers/gfx_texture#53]
Passing multiple boolean flags is bad API design. This should be replaced with a struct so the code is self documenting.

**Comment by Potpourri** [comment]
I agree, but `force_alpha` and `flip_vertical` is unnecessary in `Texture::from_image`.

**Comment by bvssvni** [comment]
@Potpourri Ah, there is a `Settings` struct already!

**Comment by bvssvni** [comment]
Maybe we should redesign `Settings` a bit. The members could be private it could use builder methods with defaults.

**Comment by Potpourri** [comment]
Yeah, I think builder is single right solution.

**Comment by kvark** [comment]
:+1:

**Comment by kvark** [comment]
Alternatively, move the flipping call into `from_image`.

**Comment by bvssvni** [comment]
I'll remove `Settings::force_alpha` since it is not implemented.
